### DISCUSSION TITLE
issue: Inconsistent value update behaviour on unmounted and remounted component

### DISCUSSION THREAD

**Discussion by rainabozhi** [react-hook-form/react-hook-form#10174] [Q&A] [answered]
### Version Number

7.43.1

### Codesandbox/Expo snack

https://codesandbox.io/s/wild-water-p14kkg?file=/src/tab-1.component.tsx

### Steps to reproduce

1. two inputs in two separate tabs & separate react hook forms share a value (fetched from our BE on initialization, requirement is also the value is updated/saved onBlur for each input)

2. on updating the input value in tab 1 and then navigating to tab 2 it is expected that the value there it also updated

3. instead, the value only updates on the first and second edit

4. upon editing the input value in tab 1 for a third time- the input value in tab 2 doesn't update although it is saved and  would display on page refresh.

I am sorry for not being able to exactly reproduce the behaviour in codesandbox - as on tab change non saved values get entirely lost. 

### Expected behaviour

Consistent value updates on every tab change 

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by leapful** [comment]
@rainabozhi 

There are several issues on your implementation on the codesanbox:

1. Your `atom` object is `{ inputValue: string }` but then when you perform update `atom` via `onBlur` callback, the field input is in string type but now you prepare method type definition is `Order`.
2. Method `reset` is used on updating atom which is not correct as you mention that you expected consistent value between all tabs.

After correcting those points, it will work as expected.

Working demo: https://codesandbox.io/s/angry-architecture-jvzrj7?file=/src/tab-1.component.tsx

**Comment by rainabozhi** [comment, marked as answer]
@leapful thank you so much for the considerate answer and the updated codesandbox. However,I am not able to reproduce my issue within the codesandbox- I was hoping the setup would give a hint to what is happening as I'm clueless there. 

The updateOrder function updates the entire object- where partially values from nested objects would populate the respective forms (which works fine). 

We use reset so that every time the user clicks inside a field the formState is updated and checked- and API calls are made only in case a value change was made. 

It works as expected- but only two out of 3 times. Meaning- one time, I would add a value, switch tabs (I see the new value in the second input as well). Second time I would add a value, switch tabs (I see the new value in the second input as well).

 Third time I would add a value, switch tabs- and now the old value is in the UI, although the API call was made and on refresh the new value was saved, just not displayed.

**Reply by rainabozhi** [reply]
Update on this- after thinking about it and discussing with the team- it seems to be an issue with data updating within the tab components that unmount. We will change our updating logic, so no issue regarding react hook forms there.
